text/0000-project-descriptor.md

```diff
@@ -0,0 +1,353 @@
+```
+
+There is an open question about how if actually run distinct builds, or if the images are derived from a single build.
+
+## Example: Codified Buildpacks
+
+Given an app with a `buildpack.toml`, the lifecycle will read the `project.buildpacks` and generate an ephemeral buildpack group in the lifecycle. Only the buildpacks listed in `project.buildpacks` will be run (no other groups will be run). For example, an app might contain:
+
+```toml
+[[project.buildpacks]]
+id = "heroku/java"
+version = "1.0"
+```
+
+These entries override any defaults in the builder image. If, for example, the project code contains a `Gemfile` and the `heroku/buildpacks` builder image is used, this will override the default buildpack groups, which would normally detect and run the `heroku/ruby` buildpack.
+
+It is also possible to define multiple buildpacks, or create an order of buildpacks that includes an inline buildpack. For example:
+
+```toml
+[[project.buildpacks]]
+id = "my-node-buildpack"
+version = "0.9"
+
+[[buildpack]]
+id = "my-node-buildpack"
+name = "My Node Buildpack"
+version = "0.9"
+[[buildpack.order]]
+group = [
+   { id = "io.buildpacks.node", version = "0.0.5" },
+   { id = "io.buildpacks.npm", version = "0.0.7" },
+]
+
+[[buildpack.dependencies]]
+id = "my-npm"
+name = "My NPM Buildpack"
+version = "0.0.7"
+path = "./npm-cnb/"
+```
+
+In this case, the inline NPM buildpack is combined with a Node engine buildpack to create a custom Node buildpack.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- This proposal merges to previously distinct concepts: projects/apps and buildpacks. However, by both concepts have many similar attributes and combining them enables the use of buildpacks to build buildpacks.
+- The use of a `buildpack.toml` to define an app may not be intuitive to a buildpack author who has already used it on a buildpack.
+
+# Alternatives
+[alternatives]: #alternatives
+
+- Name the file `project.toml` and keep the schema described herein.
+- Name the file `package.toml` and keep the schema described herein.
+- Overload `buildpack.toml` as described in [PR #2](https://github.com/buildpack/rfcs/pull/3)
+
+# Prior Art
+[prior-art]: #prior-art
+
+- [`manifest.json`](https://docs.cloudfoundry.org/devguide/deploy-apps/manifest.html)
+- [`app.json`](https://devcenter.heroku.com/articles/app-json-schema)
+- [`.buildpacks`](https://github.com/heroku/heroku-buildpack-multi)
+- [heroku-buildpack-inline](https://github.com/kr/heroku-buildpack-inline)
+- [heroku-buildpack-multi-procfile](https://github.com/heroku/heroku-buildpack-multi-procfile)
+
+# Unresolved Questions
+[unresolved-questions]: #unresolved-questions
+
+- Is `path` overriding `uri` and other elements confusing? Should these be the same key, and the execution environment figures out how to handle (i.e. `/user/local` versus `https://example.com`).
+- Do multiple `[[images]]`  result in multiple builds, or are the images derived from a single build?
```

Comment:
Thinking about this, it would be really easy to create multiple images in a single build by allowing buildpacks to add image names to layer metadata.